Update rpmrepo snapshots to 20230223 (including necessary fixes and workarounds)

yih-redhat opened the pull request:
This pull request includes:

1. Fix bug https://github.com/osbuild/osbuild-composer/issues/3309 found in update snapshot pr.
2. Add workaround for bug https://bugzilla.redhat.com/show_bug.cgi?id=2162815

achilleas-k commented on tools/fix-ansible.py:
```diff
@@ -0,0 +1,10 @@
+import os
+import sys
+
+for handle in (sys.stdin, sys.stdout, sys.stderr):
+    try:
+        fd = handle.fileno()
+    except Exception:
+        continue
+
+    os.set_blocking(fd, True)
```
I would rename the script to `ansible-blocking-io` or something similar and add a comment pointing to the ansible issue that's linked in the PR description (`#80108`).  From what I understand, this isn't a bug in anisble itself but a side-effect of a different change (either in ansible or somewhere else), so it'll be good to have some information about why we're doing this for the future.

yih-redhat replied:
Done, renamed to ansible-blocking-io.py, and added more info in it about why I am doing this.

achilleas-k commented on Schutzfile:
```diff
@@ -123,14 +123,14 @@
             "title": "updates",
             "name": "updates",
-            "baseurl": "https://rpmrepo.osbuild.org/v2/mirror/public/f37/f37-x86_64-updates-released-20221124"
+            "baseurl": "https://rpmrepo.osbuild.org/v2/mirror/public/f37/f37-x86_64-updates-released-20230223"
```
Let's revert the change in F37.  If all goes green, we can keep the changes for the other snapshots so we're testing newer versions until the F37 bug is fixed.

thozza commented on tools/provision.sh:
```diff
@@ -32,6 +32,11 @@ else
 fi
 
+# workaround for bug https://bugzilla.redhat.com/show_bug.cgi?id=2162815
+if [[ "${ID}" == "fedora" || "$VERSION_ID" == "37" ]]; then
```
You should have used `&&` instead of `||` here. The package gets now installed on any Fedora distro, which is not desired.

yih-redhat replied:
Thanks for pointing this out, my mistake, fixed.

thozza commented on tools/ansible-blocking-io.py:
```diff
@@ -0,0 +1,13 @@
+# This python script is to fix ansible error in CI test. It's not a bug of ansible, but a side-effect of a different change
```
Please add python shebang`#!/usr/bin/python3`

yih-redhat replied:
Done.

thozza commented on test/cases/ostree.sh:
```diff
@@ -686,6 +686,10 @@ ansible_private_key_file=${SSH_KEY}
 
+# Fix ansible error https://github.com/osbuild/osbuild-composer/issues/3309
+greenprint "fix stdio file non-blocking issue"
+sudo python3 /usr/libexec/osbuild-composer-test/ansible-blocking-io.py
```
nitpick: You can drop `python3` once you add shebang to the script and make it executable. This applies to all files that include this.

yih-redhat replied:
Done.

thozza reviewed (commented):
Thanks for solving all CI issues. Your work is much appreciated ❤️ 

I added a few comments. I'm also not very happy with shipping `tools/ansible-blocking-io.py` as a separate script in `/usr/libexec`, but there is no better way around this if we do not want to duplicate the code in each script (which does not make sense). I also considered if it would be nicer to set the IO to blocking using a bash script instead, so that it can be just sourced by existing test cases. But that would not improve much (if anything), so it probably does not make sense.

Happy to approve this PR once you fix the missing shebang and condition in the `provision.sh` script.

yih-redhat commented:
> Thanks for solving all CI issues. Your work is much appreciated heart
> 
> I added a few comments. I'm also not very happy with shipping `tools/ansible-blocking-io.py` as a separate script in `/usr/libexec`, but there is no better way around this if we do not want to duplicate the code in each script (which does not make sense). I also considered if it would be nicer to set the IO to blocking using a bash script instead, so that it can be just sourced by existing test cases. But that would not improve much (if anything), so it probably does not make sense.
> 
> Happy to approve this PR once you fix the missing shebang and condition in the `provision.sh` script. Please also rebase on top of `main` while making changes.

Rebased with main and pushed again.
I agree it's better to set IO to blocking using bash, I will take a look of this later.

thozza reviewed (approved):
LGTM 👍

croissanne commented:
Packer is currently broken, see #3320. Admin merge one ostree is green?
